### PR TITLE
[ENH] Clean AptaMCTS implementation and fix MCTS simulation

### LINKED ISSUE #21: Clean implementation of AptaMCTS
_UPDATED: 07-04-2025_

Add a clean implementation of AptaMCTS, an aptamer recommendation pipeline which combines RandomForest and Monthe Carlo Tree Search (MCTS). These components are the model/classifier and the string optimization algorithm, respectively.

**Todo.** Add a clean, simplified Python implementation of AptaMCTS's model/classifier and preprocessing pipeline to the repository. The MCTS algorithm is already implemented and can easily be used by model with a compatible interface (e.g., see AptaNet and AptaTrans).

High-level steps could be the following:
- Look at the authors' code, specifically the `paper_version` folder, for insights into the original implementation.
- Refactor and add to the repository the model/classifier (in `classifier.py`) and any other data pre-processing logic characteristic of AptaMCTS.
- Implement corresponding test batteries.

AptaMCTS is mostly a RandomForest classifier and leverages `sklearn`. I would expect a very similar interface to that of AptaNet, with the following classes:
- A pipeline module, `aptamcts._pipeline.py`, for the overall pipeline logic. 
- If needed, a model module, `aptamcts._feature_classifier.py`, implementing a classifier and regressor version. This module could also contain specific data preprocessing steps, if any.
- An experiment module, `experiments._aptamer_aptamcts.py`, so that AptaMCTS classifier may be combined with the current implementation of MCTS.

In the end, we want a clean AptaMCTS implementation, easily usable for training and predicting, and easily usable in combination with the current implementation of MCTS for recommending aptamer candidates.

**Code:** [https://github.com/leekh7411/Apta-MCTS](https://github.com/leekh7411/Apta-MCTS).
**Paper:** Lee, Gwangho, et al. "Predicting aptamer sequences that interact with target proteins using an aptamer-protein interaction classifier and a Monte Carlo tree search approach." PloS one 16.6 (2021): e0253760.

### PR DESCRIPTION
**Pull request by Flamki**
## Summary
This PR addresses issue #21 by adding a clean AptaMCTS implementation and fixing correctness issues in the underlying MCTS simulation logic.

### What changed
- Added a dedicated `AptaMCTS` pipeline:
  - `pyaptamer/aptamcts/__init__.py`
  - `pyaptamer/aptamcts/_pipeline.py`
  - `pyaptamer/aptamcts/tests/test_aptamcts.py`
- Fixed `MCTS` simulation correctness in `pyaptamer/mcts/_algorithm.py`:
  - token-based depth accounting (instead of raw string length)
  - simulation evaluates reconstructed aptamer sequence (consistent scoring format)
  - scalar score normalization for stable backpropagation
- Added regression coverage in `pyaptamer/mcts/tests/test_mcts.py` for reconstructed-sequence simulation and score type.

## Why
- Issue #21 requested a clean AptaMCTS implementation.
- Current MCTS simulation had a sequence-length/token mismatch and inconsistent scoring input format during simulation vs final candidate evaluation.

## Validation
- `pre-commit run --files <changed files>`
- `python -m pytest pyaptamer\\mcts\\tests\\test_mcts.py pyaptamer\\aptamcts\\tests\\test_aptamcts.py -p no:warnings`
- `python -m pytest pyaptamer\\aptatrans\\tests\\test_aptatrans.py pyaptamer\\experiments\\tests\\test_aptamer.py -p no:warnings`
- `python -m pytest pyaptamer -p no:warnings`

Closes #21.

### CODE REVIEW

**Comment by siddharth7113** [comment]
Hi @Flamki ,

We are currently not engaging this PR due to heavy AI usage suspected, and with large diff , the burden falls on reviewer, we are still discussing on AI policy, while we allow AI assisted development, we are unable to review complete AI generated PRs.

I would request you to work good first issues, especially ones that are `easy` and `medium` tagged .
